test/e2e/reconciler/partition/partitionset_test.go

```diff
@@ -0,0 +1,332 @@
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/kcp-dev/kcp/pkg/apis/core"
+	corev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+	topologyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/topology/v1alpha1"
+	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/cluster"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestPartitionSet(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	server := framework.PrivateKcpServer(t)
+
+	// Create organization and workspace
+	orgPath, _ := framework.NewOrganizationFixture(t, server)
+	partitionClusterPath, _ := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("partitionset"))
+
+	cfg := server.BaseConfig(t)
+	kcpClusterClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+	partitionSetClient := kcpClusterClient.TopologyV1alpha1().PartitionSets()
+
+	t.Logf("Creating a partitionSet not matching any shard")
+	partitionSet := &topologyv1alpha1.PartitionSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-partitionset",
+		},
+		Spec: topologyv1alpha1.PartitionSetSpec{
+			Dimensions: []string{"region"},
+			ShardSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      "excluded",
+						Operator: metav1.LabelSelectorOpDoesNotExist,
+					},
+				},
+			},
+		},
+	}
+	partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Create(ctx, partitionSet, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating partitionSet")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionSetValid) && conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) {
+			return true, ""
+		}
+		return false, spew.Sdump(partitionSet.Status.Conditions)
+	}, wait.ForeverTestTimeout, 100*time.Millisecond, "expected valid partitionSet")
+
+	t.Logf("Creating a shard matching the partitionSet")
+	shard1a := &corev1alpha1.Shard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "partition-shard-1a",
+			Labels: map[string]string{
+				"region": "partition-test-region-1",
+			},
+		},
+		Spec: corev1alpha1.ShardSpec{
+			BaseURL: "https://base.kcp.test.dev",
+		},
+	}
+	shardClient := kcpClusterClient.CoreV1alpha1().Shards()
+	_, err = shardClient.Cluster(core.RootCluster.Path()).Create(ctx, shard1a, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating shard")
+	framework.Eventually(t, func() (bool, string) {
```

Comment:
```
framework.EventuallyCondition
```

Reply:
did not exist when this test was written, does not allow to check values of multiple conditions.